app/common/auth/__init__.py

```
@@ -94,6 +96,12 @@ def sso_get_token() -> ResponseReturnValue:
 
     user = get_or_create_user(email_address=sso_user["preferred_username"])
 
+    if "FSD_ADMIN" in sso_user["roles"]:
+        try:
+            add_user_role(user_id=user.id, role=RoleEnum.ADMIN)
+        except InvalidUserRoleError as e:
+            abort(400, description=str(e))
```

I'm not sure we want to catch this error, do we?

The user isn't in control of the role that gets added, so a 400 (client error) feels inappropriate - there's nothing they can do to fix it?

We should probably just not catch the error and let it bubble up to a 500.

---

Yeah the error won't ever really get surfaced here as we're always passing a correct combo, but the thinking behind adding it here was as a bit of an eg. for future uses of the `add_user_role` interface - happy to remove it and just raise the error though.